test(rbno): Add longevities for repair based node operations

roydahan opened the pull request:
Adding several longevities to test RBNO feature.
Four types of nemesis are being used to test it:
1. Remove Node and bootstrap new node instead.
2. Replace a dead node.
3. Decommission and add new node.
4. Grow cluster by adding nodes then shrink it back with decommission

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

roydahan commented:
@juliayakovlev this is the PR with all the RBNO jobs.
For now you need to use only those for "ReplaceNode".
I suggest you will run 1-2 of them in staging, making sure they work and then we merge them and run them in 4.6.

roydahan commented:
I need to change the longevities to use only the "incremental" changes in the configuration.

roydahan commented:
@juliayakovlev I changed all yamls to be "incremental" (only changing what needed on top of the original longevity).
Also, I changed the pipelines accordingly.

Please test RBNO feature for replace node with it.
Run it once from this branch, if it's ok, we will merge and you can run the official jobs from official branch.

juliayakovlev commented on test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:
```diff
@@ -0,0 +1,5 @@
+nemesis_class_name: 'NodeTerminateAndReplace'
+nemesis_interval: 2
+nemesis_filter_seeds: false
```
@roydahan Why did you set `nemesis_filter_seeds` to false? "longevity-50GB-3days-authorization-and-tls-ssl.yaml" set 1 seed node only

roydahan replied:
I don't remember, but if it's a problem we can set to true or better increase the number of seeds.

juliayakovlev replied:
Yes, we can't run terminate and replace on seed if one seed in the cluster only.
I created my branch, set seed_num to 3 and run the test with my branch. Can you, please, fix it here

roydahan replied:
you can push to this PR

juliayakovlev commented:
@roydahan Roy, what is test duration should be? You called the tests "4h", but actually the test duration set in`longevity-50GB-3days-authorization-and-tls-ssl`, that 3 days

roydahan commented:
> @roydahan Roy, what is test duration should be? You called the tests "4h", but actually the test duration set in`longevity-50GB-3days-authorization-and-tls-ssl`, that 3 days

where?

bentsi commented on test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml:
```diff
@@ -0,0 +1,6 @@
+nemesis_class_name: 'TerminateAndRemoveNodeMonkey'
```
missing `test_duration: 240`

bentsi commented on jenkins-pipelines/repair-based-operation/longevity-cdc-4h-ReplaceNode.jenkinsfile:
```diff
@@ -0,0 +1,13 @@
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml", "test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-ReplaceNode.yaml"]''',
+
+    timeout: [time: 360, unit: 'MINUTES']
```
not needed

roydahan commented on jenkins-pipelines/repair-based-operation/longevity-100gb-4h-ReplaceNode.jenkinsfile:
```diff
@@ -0,0 +1,13 @@
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-50GB-3days-authorization-and-tls-ssl.yaml", "test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml"]''',
+
+    timeout: [time: 360, unit: 'MINUTES']
```
This is how almost all our pipeline looks like.
If we want we need to send a PR and remove from all of them.

fgelcer replied:
yes, it need to be removed from all `jenkinsfile` as it is not used anymore

roydahan commented on test-cases/features/repair-based-operations/longevity-100gb-4h-ReplaceNode.yaml:
```diff
@@ -0,0 +1,5 @@
+nemesis_class_name: 'NodeTerminateAndReplace'
```
The test_duration is in the "base" yaml (each pipeline has two yamls).
This yaml is only overriding the base yamls with some parameters.
There was a mistake in the base yaml that used incorrect yaml of a longer run, now it's fixed.

roydahan commented:
@bentsi & @juliayakovlev, I added "seed_num: 3" to the "ReplaceNode" longevities that has "nemesis_filter_seed: false"
It should do.
I also fixed a mistake where some of the pipelines used the wrong base yaml.

juliayakovlev commented on test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-RemoveNode.yaml:
```diff
@@ -0,0 +1,6 @@
+nemesis_class_name: 'TerminateAndRemoveNodeMonkey'
+nemesis_interval: 5
+nemesis_during_prepare: false
```
Please, add `seeds_num` or chande `nemesis_filter_seeds` to True
"longevity-cdc-100gb-4h.yaml" runs with 1 seed.

roydahan replied:
Same as above.

roydahan commented on test-cases/features/repair-based-operations/longevity-cdc-100gb-4h-Decommission.yaml:
```diff
@@ -0,0 +1,6 @@
+nemesis_class_name: 'DecommissionMonkey'
+nemesis_interval: 5
+nemesis_during_prepare: false
```
1. Why is it needed here?
2. We don't care now about Decommission operation.

juliayakovlev replied:
my mistake

roydahan commented on test-cases/features/repair-based-operations/longevity-50GB-12h-ReplaceNode.yaml:
```diff
@@ -0,0 +1,12 @@
+
+nemesis_class_name: 'NodeTerminateAndReplace'
+nemesis_interval: 5
+nemesis_during_prepare: false
```
we don't have  nemesis_filter_seeds here.

juliayakovlev replied:
my mistake

bentsi commented:
@roydahan please fix pre-commit
![image](https://user-images.githubusercontent.com/5269241/155889450-2c4fee15-6b69-49d7-b610-a9d237a4fa70.png)

roydahan commented:
@bentsi passed all checks
